x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -706,12 +706,20 @@ class AgentPolicyService {
       } catch (error) {
         throw new Error('Default settings is not setup');
       }
-      if (!settings.kibana_urls || !settings.kibana_urls.length)
-        throw new Error('kibana_urls is missing');
-
-      fullAgentPolicy.fleet = {
-        kibana: getFullAgentPolicyKibanaConfig(settings.kibana_urls),
-      };
+      if (settings.fleet_server_urls && settings.fleet_server_urls.length) {
+        fullAgentPolicy.fleet = {
+          hosts: settings.fleet_server_urls,
+        };
+      } // TODO remove as part of https://github.com/elastic/kibana/issues/94303
+      else {
+        if (!settings.kibana_urls || !settings.kibana_urls.length)
+          throw new Error('kibana_urls is missing');
+
+        fullAgentPolicy.fleet = {
+          hosts: settings.kibana_urls,
```

Comment:
is it expected to have `fleet.hosts` be kibana URLs when there there aren't any fleet server urls configured yet? i guess this is temporary until we remove kibana URLs support?

Reply:
Yes that correct agent 7.13 will only support `fleet.hosts` even for Kibana during a short period to not break all of our CI